internal/mode/static/nginx/config/stream_servers.go

```diff
@@ -0,0 +1,56 @@
+package config
+
+import (
+	"fmt"
+	gotemplate "text/template"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/helpers"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/config/stream"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/dataplane"
+)
+
+var streamServersTemplate = gotemplate.Must(gotemplate.New("servers").Parse(streamServersTemplateText))
+
+func executeStreamServers(conf dataplane.Configuration) []executeResult {
+	streamServers := createStreamServers(conf)
+
+	streamServerResult := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(streamServersTemplate, streamServers),
+	}
+
+	result := []executeResult{
+		streamServerResult,
+	}
+
+	return result
+}
+
+func createStreamServers(conf dataplane.Configuration) []stream.Server {
+	streamServers := make([]stream.Server, 0)
+	for _, t := range conf.TLSServers {
```

Comment:
What if we only have one TLSServer? Or if every TLSServer is listening on a different port? Can we simplify in that case?

Reply:
I don't understand this comment. What is needed to simplify?

Reply:
We don't need a map if there's only one TLSServer, or if every TLSServer is listening on a different port.